.pre-commit-config.yaml

```diff
@@ -0,0 +1,24 @@
+repos:
+  - repo: local
+    hooks:
+      - id: black
+        name: format python with black
+        language: system
+        entry: docker-compose exec -T web black
+        types: [python]
+      - id: prettier
+        name: format javascript with prettier
+        language: system
+        entry: docker-compose exec -T web $(shell npm $(NPM_ARGS) bin)/prettier
+        args: [--write]
+        types: [javascript]
+      - id: flake8
+        name: lint python with flake8
+        language: system
+        entry: docker-compose exec -T web flake8
+        types: [python]
+      - id: curlylint
+        name: lint html with curlylint
+        language: system
+        entry: docker-compose exec -T web curlylint
+        types: [html]
```

Comment:
would `docker-compose exec -T web make format` then `docker-compose exec -T web make lint` replace this?  (It'd be consistent)

Reply:
The advantage of pre-commit is being is being able to easily share the hooks. It's not possible to add .git/hooks to a repo unfortunately, the workaround is configuring a custom path and locally updating git config.

It is possible to call `make lint` and `make format` although they process the entire repo rather than just the files which are being committed, which is time consuming.

I could add optional args to lint and format in makefile-docker, and the pre-commit hook would pass the changed filenames?